### PR TITLE
Add pybind11-rdp 0.1.1 recipe

### PR DESCRIPTION
**Pull request by district10**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by district10** [comment]
@conda-forge/help-c-cpp I'm using pybind11 to bind a c++ implementation of Ramer-Douglas-Peucker algorithm, source code at: https://github.com/cubao/pybind11-rdp.

I have trouble fix these CI failures, any ideas?

I read this from https://github.com/pybind/cmake_example:

> An example [pybind11](https://github.com/pybind/pybind11) module built with a CMake-based build system. This is useful for C++ codebases that have an existing CMake project structure. This is in many cases superseded by [scikit_build_example](https://github.com/pybind/scikit_build_example), which uses [scikit-build](https://scikit-build.readthedocs.io/en/latest/), a tool from the makers of CMake designed to allow Python packages to be driven from CMake. However, there are still cases where you might want full control over the CMake run; and both of these approaches have some trade-offs not present in a pure setuptools build (see [python_example](https://github.com/pybind/python_example)). Python 3.7+ required; see the commit history for older versions of Python.

I'm supposed to use scikit_build instead?

**Comment by district10** [comment]
@conda-forge-admin

File: recipes/pybind11-rdp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package:
+  name: pybind11-rdp
+  version: 0.1.1
+
+source:
+  url: https://files.pythonhosted.org/packages/57/f3/ba8605feae325db6d493ba604d5a07e08ff3092e83024799966986dc0c23/pybind11_rdp-0.1.1.tar.gz
```
Comment (carterbox) [inline recipes/pybind11-rdp/meta.yaml:6]: Please use the release URL as shown in our example recipe.

https://github.com/conda-forge/staged-recipes/blob/43cb020eb60984df338fd750d7a6ba1ecf3c4257/recipes/example/meta.yaml#L16

File: recipes/pybind11-rdp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  sha256: 9d39d4054fb1d3d2f1ab60252575dd1be068f1b9d27133ee289de6aef1ddde98
+
+build:
+  number: 0
+  script: python -m pip install . -vvv
+  skip: True  # [win]
+
+requirements:
+  build:
+    - "{{ compiler('cxx') }}"
```
Comment (carterbox) [inline recipes/pybind11-rdp/meta.yaml:16]:
```suggestion
    - {{ compiler('cxx') }}
```

File: recipes/pybind11-rdp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - cmake
+    - make  # [not win]
+    - ninja
+
+  host:
+    - python
+    - pip !=22.1.0
```
Comment (carterbox) [inline recipes/pybind11-rdp/meta.yaml:23]: Missing some packages from host.

https://github.com/cubao/pybind11-rdp/blob/master/pyproject.toml

File: recipes/pybind11-rdp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+test:
+  requires:
+    - pytest
+    - numpy
+  imports:
+    - pybind11_rdp
+  source_files:
+    - tests
+  commands:
+    - python -m pytest
```
Comment (carterbox) [inline recipes/pybind11-rdp/meta.yaml:37]:
```suggestion
test:
  requires:
    - pytest
    - pip
  imports:
    - pybind11_rdp
  source_files:
    - tests
  commands:
    - pip check
    - python -m pytest 
```

File: recipes/pybind11-rdp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    - cmake
+    - make  # [not win]
+    - ninja
+
+  host:
+    - python
+    - pip !=22.1.0
+
+  run:
+    - python
```
Comment (carterbox) [inline recipes/pybind11-rdp/meta.yaml:26]:
```suggestion
  run:
    - python
    - numpy
```
Missing a package from run.

https://github.com/cubao/pybind11-rdp/blob/a2c80310aed756d39f8365d91b62800db8d9d0e3/setup.py#L135

**Review by carterbox — CHANGES_REQUESTED** [review]
`pybind11-rdp` is not a good package name because it implies an extension of the pybind11 package instead of an implementation of RDP using pybind.

Have you considered opening a pull request [at the upstream package](https://github.com/fhirschmann/rdp/pulls)?


> Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).

This package (`pybind11-rdp`) is revending multiple other packages including Pybind11, Eigen, spdlog, rapidjson, etc. As the checklist mentions, we do not allow revending of other packages. Please use the conda-forge versioned releases of these packages or explain why you cannot.

Also, `pybind11-rdp` does not have a license of its own. That is also required.

**Comment by carterbox** [comment]
```
-- Building for: NMake Makefiles
  CMake Error at CMakeLists.txt:2 (project):
    Generator

      NMake Makefiles

    does not support platform specification, but platform

      x64

    was specified.
```
This is the build error for Windows. It looks like incorrect arguments are being passed to NMake from `setup.py`.

**Pull request closed without merging**